content_scripts/clipboard.js

```diff
@@ -32,6 +36,11 @@ function createClipboard() {
     };
 
     self.write = function(text) {
+        const cb = () => Front.showBanner("Copied: " + text);
+        if (typeof navigator.clipboard === 'object' && typeof navigator.clipboard.writeText === 'function') {
+          navigator.clipboard.writeText(text).then(cb);
```

Comment:
I'm fairly sure this won't work on non-HTTPS pages - see https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Interact_with_the_clipboard - has anyone checked? : )

Reply:
Thanks for mentioning this, it actually did not cross my mind to test that. I will double check when I get a chance later today.

Reply:
Just got around to testing this and you are absolutely correct, @bovine3dom. 

Since @brookhong added a check in 116e52af5f552920db7582adc0272d8a8b086b5f to only enable this behavior if the browser is Firefox, at least there shouldn't be any regression in version 84 onward since it was totally broken. Until we can determine a better path forward that works on insecure sites as well, I think this should suffice most use cases.

Perhaps an additional check should be put in place to check the browser version is 84 or greater so that users on old versions don't have this functionality broken on insecure sites?

Reply:
All you need to do in Firefox is execute the `clipboard` functions from the background script rather than content.

Surfingkeys will already be doing that sort of thing in lots of other places so it should be quite straightforward to send it a message from here and get the result back.